replace boost dependency with rapidxml header

With guidance from @jeroenooms I uninstalled boost from my system to test portability:

```
sudo apt-get purge libboost-all-dev 
sudo apt-get autoremove
```

Then tried the following:

``` r
devtools::install_github("osmdatar/osmdatar") # error
```

Then tried the following (after adding the **BH** link):

``` r
devtools::install_github("robinlovelace/osmdatar") # works
```

Full error message from 1st option:

```
devtools::install_github("osmdatar/osmdatar")
Downloading GitHub repo osmdatar/osmdatar@master
from URL https://api.github.com/repos/osmdatar/osmdatar/zipball/master
Installing osmdatar
'/usr/lib/R/bin/R' --no-site-file --no-environ --no-save --no-restore --quiet CMD  \
  INSTALL '/tmp/RtmpAGPJnQ/devtools71b86445042/osmdatar-osmdatar-891d23c'  \
  --library='/home/robin/R/x86_64-pc-linux-gnu-library/3.3' --install-tests 

* installing *source* package ‘osmdatar’ ...
** libs
g++ -std=c++11 -I/usr/share/R/include -DNDEBUG -fprofile-arcs -ftest-coverage  -I"/home/robin/R/x86_64-pc-linux-gnu-library/3.3/Rcpp/include"   -fpic  -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g -c RcppExports.cpp -o RcppExports.o
g++ -std=c++11 -I/usr/share/R/include -DNDEBUG -fprofile-arcs -ftest-coverage  -I"/home/robin/R/x86_64-pc-linux-gnu-library/3.3/Rcpp/include"   -fpic  -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g -c common.cpp -o common.o
In file included from common.cpp:2:0:
common.h:3:41: fatal error: boost/property_tree/ptree.hpp: No such file or directory
compilation terminated.
/usr/lib/R/etc/Makeconf:141: recipe for target 'common.o' failed
make: *** [common.o] Error 1
ERROR: compilation failed for package ‘osmdatar’
* removing ‘/home/robin/R/x86_64-pc-linux-gnu-library/3.3/osmdatar’
* restoring previous ‘/home/robin/R/x86_64-pc-linux-gnu-library/3.3/osmdatar’
Error: Command failed (1)
```

--- a/DESCRIPTION
+++ b/DESCRIPTION
@@ -26,7 +26,7 @@ Suggests:
     roxygen2,
     rmarkdown,
     testthat
-LinkingTo: Rcpp
+LinkingTo: Rcpp, BH
 URL: https://github.com/osmdatar/osmdatar
 BugReports: https://github.com/osmdatar/osmdatar/issues
 Date: 2016-08-23

